sklearn/utils/estimator_checks.py

```diff
@@ -2971,3 +2989,30 @@ def check_n_features_in(name, estimator_orig):
             "https://scikit-learn-enhancement-proposals.readthedocs.io/en/latest/slep010/proposal.html",  # noqa
             FutureWarning
         )
+
+
+def skip_if_not_strict(check):
+    # Decorator to skip check when their strict_mode parameter is False
+    @wraps(check)
+    def wrapper(*args, **kwargs):
+        strict_mode = kwargs.get('strict_mode', True)
+        if strict_mode:
+            return check(*args, **kwargs)
+        else:
+            raise SkipTest(f"Skipping {check} since strict mode is off")
```

Comment:
I think the issue may be that this decorator shouldn't raise skip test (which might not be handled well in a pytest session), but rather only mark the test as skipped in some way. And we can skip it in the same place we currently mark tests as xfail. 

I'll try to have a more detailed look in a couple of days..